sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -200,6 +200,8 @@ object RemoveAliasOnlyProject extends Rule[LogicalPlan] {
         case plan: Project if plan eq proj => plan.child
```

Comment:
a naive way to do this is, make sure we only replace attributes in the ancestor nodes of the alias-only project:
```
plan transform {
  case plan: Project if plan eq proj => plan.child
  case plan if plan.collect { case p if p eq project }.nonEmpty => // do the replace
}
```

It's very inefficient, maybe we can improve `TreeNode` to maintain the parent-child relationship between nodes.